File: python-examples/quickstart/consumer.py
#!/usr/bin/env python3
"""
Pulsar Python Consumer 快速入门示例
对应文章: 01-入门篇/01-5分钟上手Pulsar.md
"""

import pulsar

def main():
    # 创建 Pulsar 客户端
    client = pulsar.Client('pulsar://localhost:6650')
    
    # 创建消费者
    consumer = client.subscribe('my-python-topic', 'python-sub')
    
    try:
        # 接收消息
        msg = consumer.receive()
        print("📩 收到消息:", msg.data().decode('utf-8'))
        
        # 确认消息
        consumer.acknowledge(msg)
    finally:
        # 关闭资源
        consumer.close()
        client.close()

if __name__ == "__main__":
    main()

File: go-examples/quickstart/consumer.go
package main

import (
	"context"
	"fmt"
	"log"

	"github.com/apache/pulsar-client-go/pulsar"
)

func main() {
	// 创建 Pulsar 客户端
	client, err := pulsar.NewClient(pulsar.ClientOptions{
		URL: "pulsar://localhost:6650",
	})
	if err != nil {
		log.Fatal(err)
	}
	defer client.Close()

	fmt.Println("🚀 开始 Consumer 示例...\n")

	// 创建 Consumer
	consumer, err := client.Subscribe(pulsar.ConsumerOptions{
		Topic:            "persistent://public/default/go-quickstart-topic",
		SubscriptionName: "go-quickstart-subscription",
		Type:             pulsar.Exclusive,
	})
	if err != nil {
		log.Fatal(err)
	}
	defer consumer.Close()

	fmt.Println("✅ Consumer 已创建")
	fmt.Println("💡 等待接收消息（按 Ctrl+C 退出）...\n")

	// 接收消息
	fmt.Println("📩 接收消息:")
	for i := 0; i < 5; i++ {
		msg, err := consumer.Receive(context.Background())
		if err != nil {
			log.Fatal(err)
		}

		fmt.Printf("   ✅ 接收: %s\n", string(msg.Payload()))
		fmt.Printf("      消息ID: %v\n", msg.ID())

		// 确认消息
		consumer.Ack(msg)
	}

	fmt.Println("\n✅ Consumer 示例执行完成")
}
